client/src/components/ratingsReviews/RatingsAndReviews.jsx

```diff
@@ -217,21 +220,59 @@ class RatingsAndReviews extends React.Component {
     });
   }
 
+  updateRatingsList(filteredList) {
+    this.setState({
+      filteredRatings: filteredList
+    }, () => {
+
+      console.log('TEST1:', this.state.filteredRatings);
+    });
+  }
+
   handleRatingProgressFilter(event) {
-    event.preventDefault();
-    let starFilter = parseInt(event.target.id);
-    console.log('TEST3', this.state[starFilter]);
-    this.state[starFilter] ?
+    let filteredRatings = [];
+    let starFilter = event.target.id;
+
+    if (this.state[starFilter]) {
+      console.log('Star Filter is true');
       this.setState({
-        starFilter: false
+        [starFilter]: false
       }, () => {
-        this.getReviews(this.state.sort, this.state.count, this.props.selectedProduct.id);
-      })
-      : this.setState({
-        starFilter: true
+        console.log('Ratings before cut:', filteredRatings);
+        this.state.reviews.forEach((review) => {
+          const rating = review.rating;
+
+          if (rating === parseInt(starFilter)) {
+            filteredRatings.splice(filteredRatings.indexOf(review), 1);
+            console.log('Cut out former ratings:', filteredRatings);
+          }
+        });
+        if (filteredRatings.length === 0) {
+          filteredRatings === null;
```

Comment:
WTF? Is this supposed to be an assignment?

Reply:
Yes, I updated it. Thank you!